aea/registries/base.py

```diff
@@ -728,16 +729,32 @@ def decision_maker_out_queue(self) -> Queue:
         """Get decision maker (out) queue."""
         return self._decision_maker_out_queue
 
-    def get_active_handlers(self, protocol_id: PublicId) -> List[Handler]:
+    def get_active_handlers(
+        self, protocol_id: PublicId, envelope_context: Optional[EnvelopeContext]
+    ) -> List[Handler]:
         """
         Get active handlers.
 
         :param protocol_id: the protocol id
+        :param envelope context: the envelope context
         :return: the list of handlers currently active
         """
-        # TODO naive implementation
-        handlers = self.resources.handler_registry.fetch_by_protocol(protocol_id)
-        active_handlers = list(filter(lambda h: h.context.is_active, handlers))
+        if envelope_context is not None and envelope_context.uri is not None:
+            uri_path = envelope_context.uri.path
+            skill_id = PublicId.from_str(uri_path)
```

Comment:
@marcofavorito we have this uri available in the envelope context. In the HTTP connection this will be filled with the request URL. It would be nice if we could use the URI to map to a skill. But our skill ids are not compatible with the path (`:`) unless we do something like `fetchai/echo/0.1.0` mapping to `fetchai/echo:0.1.0`. Thoughts?

Reply:
I think it is not too bad to have `fetchai/echo/0.1.0` as URI.
We could add methods like `public_id.to_uri()` and `PublicId.from_uri()` to parse them.